### PR TITLE
Introduce the splunk-otel-java-agent framework.

### PR DESCRIPTION
**Pull request by breedx-splk**
This framework will wire up the Splunk distribution of OpenTelemetry java for auto-instrumentation and direct ingest. It leverages VCAP_SERVICES and the existence of a service binding with a specific name (splunk-o11y).

Signed-off-by: Jason Plumb jplumb@splunk.com

Resolves issue #825

### CODE REVIEW

**Review by dmikusa — COMMENTED** [review]
Thanks for the PR, much appreciated! A few bits of feedback.

File: docs/framework-splunk_otel_java_agent.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Splunk Distribution of OpenTelemetry Java Instrumentation
+
+The Splunk OpenTelemetry Java Agent buildpack framework will cause an application to be automatically instrumented
+with the [Splunk distribution of OpenTelemetry Java Instrumentation](https://github.com/signalfx/splunk-otel-java).
+
+Trace data will be sent directly to Splunk Observability Cloud. 
+
+* **Detection criteria**: Existence of a bound service with the name `splunk-o11y`.
```
[the remark this answers is not in the document]
Comment (breedx-splk) [reply docs/framework-splunk_otel_java_agent.md:8]: Yeah, I hadn't considered the dev/prod environment thing, that's a good point. I think we can relax it to just containing `splunk-o11y`.

File: docs/framework-splunk_otel_java_agent.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+* **Tags**: `splunk-otel-java-agent=<version>` to control which version of the instrumentation agent is used.
```
[the remark this answers is not in the document]
Comment (breedx-splk) [reply docs/framework-splunk_otel_java_agent.md:9]: Sure thing.

File: lib/java_buildpack/framework/splunk_otel_java_agent.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        java_opts.add_javaagent(@droplet.sandbox + jar_name)
+
+        credentials = @application.services.find_service(REQUIRED_SERVICE_NAME_FILTER)['credentials']
+        # Add all otel.* and splunk.* credentials from the service bind as jvm system properties
+        credentials&.each do |key, value|
+          java_opts.add_system_property(key, value) if key.start_with?('splunk.') || key.start_with?('otel.')
+        end
+
+        app_name = @application.details['application_name']
+        java_opts.add_system_property('otel.service.name', app_name)
```
[the remark this answers is not in the document]
Comment (breedx-splk) [reply lib/java_buildpack/framework/splunk_otel_java_agent.rb:44]: Good call. I like the flexibility that would provide.

**Comment by breedx-splk** [comment]
Thanks @dmikusa for the review...greatly appreciated. I made some changes and am ready for another look.

**Comment by breedx-splk** [comment]
Hi @dmikusa Just checking back in after a few days. Thanks.

**Comment by dmikusa** [comment]
Apologizes. I don't have access to CF environments at the moment. @pivotal-david-osullivan would need to provide final review & merge.

**Comment by breedx-splk** [comment]
Hi @pivotal-david-osullivan any chance to get a review on this please? 🙏🏻

**Comment by pivotal-david-osullivan** [comment]
Apologies for the delay, I had some issues getting environments too, I will review and update ASAP!

**Comment by breedx-splk** [comment]
@pivotal-david-osullivan Just touching base again. 🤞🏻 Thanks.

**Comment by breedx-splk** [comment]
Nudging this again to keep this alive.....thanks in advance.

**Comment by breedx-splk** [comment]
Maybe we could get a look before the year is over?

**Comment by pivotal-david-osullivan** [comment]
I was able to get a lab and test this, the agent downloads and I see some initial output. Thank you for your patience on this @breedx-splk and thanks again for the PR, I've merged it in!

**Pull request merged**